bokeh/util/version.py

```diff
@@ -83,5 +83,5 @@ def _base_version_helper(version: str) -> str:
 # Code
 #-----------------------------------------------------------------------------
 
-__version__: str = get_versions()['version']
+__version__: str = get_versions()['version']  # type: ignore
```

Comment:
Not sure what is going on here, but _versions is generated and untyped, maybe newer versions of versioneer will be typed.

Reply:
See `bokeh/_version.pyi`. This should type fine. In fact explicit `: str` shouldn't be needed as well. This needs to be investigated before any ignores are applied.

Reply:
Ah I missed that there was a `.pyi` For some reason it seems to think the return is optional
```
bokeh/util/version.py:86:20: error: Value of type Versions? is not indexable  [index]
```

Reply:
I have no idea why it thinks the return is nullable 🤷

Reply:
Seems to be a bug in mypy resulting in a different interpretation of `TypedDict("X", ...)` and `class X(TypedDict): ...` definitions. This "fixes" the problem:
```diff
diff --git a/bokeh/_version.pyi b/bokeh/_version.pyi
index 27e76123b..7cdf454c0 100644
--- a/bokeh/_version.pyi
+++ b/bokeh/_version.pyi
@@ -1,12 +1,10 @@
 # Standard library imports
 from typing import TypedDict
 
-Versions = TypedDict("Versions", {
-  "version": str,
-  "full-revisionid": str | None,
-  "dirty": bool | None,
-  "error": str | None,
-  "date": str | None,
-})
+class Versions(TypedDict):
+  version: str
+  dirty: bool | None
+  error: str | None
+  date: str | None
 
 def get_versions() -> Versions: ...
diff --git a/bokeh/util/version.py b/bokeh/util/version.py
index 8524015d2..b85575892 100644
--- a/bokeh/util/version.py
+++ b/bokeh/util/version.py
@@ -83,5 +83,5 @@ def _base_version_helper(version: str) -> str:
 # Code
 #-----------------------------------------------------------------------------
 
-__version__: str = get_versions()['version']  # type: ignore
+__version__ = get_versions()['version']
 del get_versions
```
The original was there to support `-` in a field name, but given it's not being used, then we can skip it.

Reply:
I'm guessing the `: str` type was a side effect of this issue as well. Perhaps this issue presented itself different in 0.900 vs 0.910.